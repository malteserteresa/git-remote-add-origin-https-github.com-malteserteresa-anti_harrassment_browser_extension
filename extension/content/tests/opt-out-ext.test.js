const jsdom = require('jsdom');

describe('opt-out-ext.js', () => {
  let newWindow;
  beforeAll(() => {
    // Mock out MutationObserver
    global.MutationObserver = class {
      constructor (callback) { return true; }
      disconnect () {}
      observe (element, initObject) {}
    };

    // Create page
    newWindow = new jsdom.JSDOM(`
      <!DOCTYPE html>
      <html>
        <body>
          <div data-testid="tweet">
            <h1>Oh ya!</h1>
          </div>
          <div data-testid="tweet" class="processed-true">
            <h1>Oh ya!</h1>
          </div>
        </body>
      </html>
    `, {
      resources: 'usable',
      runScripts: 'dangerously'
    }).window;
    newWindow.eval(require('../opt-out-ext.js'));
  });
  describe('browser.runtime.onMessage.addListener event listener', () => {
    describe('message created from popup', () => {
      it('when new text format option configured, applies the correct class to tweets', () => {
        // Send message
        browser.runtime.sendMessage({
          selector: 'text_removed',
          slider: '1'
        });
        // Check that document.body HTML has changed some tweets
        expect(newWindow.document.body).toMatchSnapshot();
      });
    });
  });
});
