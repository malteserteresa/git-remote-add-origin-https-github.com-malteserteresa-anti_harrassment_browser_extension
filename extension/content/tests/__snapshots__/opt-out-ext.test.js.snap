// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opt-out-ext.js browser.runtime.onMessage.addListener event listener message created from popup when new text format option configured, applies the correct class to tweets 1`] = `
<body>
  
          
  <div
    data-testid="tweet"
  >
    
            
    <h1>
      Oh ya!
    </h1>
    
          
  </div>
  
          
  <div
    class="processed-true"
    data-testid="tweet"
  >
    
            
    <h1>
      Oh ya!
    </h1>
    
          
  </div>
  
        
      
    
</body>
`;
